book/src/filters.md

```diff
@@ -415,6 +415,18 @@ Ugly: <script>var data = "{{data|json}}";</script>
 Ugly: <script>var data = '{{data|json|safe}}';</script>
 ```
 
+By default, a compact representation of the data is generated, i.e. no whitespaces are generated
```

Comment:
I'd still prefer to stick with pretty by default.

Reply:
IMO, that's a bad default. The cases when you want to look at the generated JSON data will be few, esp. because you cannot specify an initial indentation level. And if you need it, why are two characters a good default? Why not four? Why not a tab character? Why not hot dogs?

[![](https://archive.ph/xwm35/ffda33094fbc849d2af86431b8eb378bb816353d.jpg)](https://archive.ph/xwm35)

Reply:
FWIW as a user I've used the json filter exclusively to assign js variables, so I think unpretty is a perfectly sane default.